isaric/parsers/ncov-malaysia.toml

```diff
@@ -0,0 +1,878 @@
+  fields = [
+    { field = "o_mx_medication_type" }, 
+    { field = "o_intervene_dc" }
+  ]
+  
+  [visit.treatment_delirium]
+    field = "mx_delirium"
+    ref = "Y/N"
+
+  [visit.treatment_delirium_type]
+    combinedType = "set"
+    excludeWhen = "none"
+    fields = [
+      { field = "mx_delirium_type___1", values = { 1 = "Haloperidol" } },
+      { field = "mx_delirium_type___1", values = { 1 = "Olanzapine" } },
+      { field = "mx_delirium_type___1", values = { 1 = "Quetipine" } },
+      { field = "mx_delirium_type___1", values = { 1 = "other" } },
+      { field = "o_mx_delirium_type" },
+    ]
+
+  [visit.treatment_oxygen_mechanical_support] # NOT SURE ABOUT THIS ONE!
+    field = "daily_case_ill_class"
+    values = { 2 = true }
+
+  [visit.outcome]
+    field = "type_outcome"
+    description = "Hospital Outcome"
+
+    [visit.outcome.values]
+      1 = "discharged"
+      2 = "hospitalised"
+      3 = "transferred"
+      4 = "death"
+      5 = "palliative discharge"
+      6 = "transferred"  # 6 = quarantined, which is not in schema currently
+      7 = "transferred"
+
+[visit.date_outcome]
+  field = "outcome_date"
+
+[[observation]]
+  name = "abdominal_pain"
+  phase = "admission"
+  date = { ref = "admissionDateHierarchy" }
+  is_present = { field = "abd_pain_hx", description = "Abdominal pain", ref = "Y/N" }
+
+[[observation]]
+  name = "base_excess"
+  phase = "study"
+  date = { field = "daily_assess_date" }
+
+  [observation.value]
+    field = "daily_baseex_data"
+
+[[observation]]
+  name = "bleeding_haemorrhage"
+  phase = "admission"
+  date = { ref = "admissionDateHierarchy" }
+
+  [observation.is_present]
+    field = "bleed_hx"
+    description = "Bleeding (Haemorrhage)"
+    ref = "Y/N"
+
+[[observation]]
+  name = "chest_pain"
+  phase = "admission"
+  date = { ref = "admissionDateHierarchy" }
+
+  [observation.is_present]
+    field = "chest_pain_hx"
+    description = "Chest pain"
+    ref = "Y/N"
+
+[[observation]]
+  name = "conjunctivitis"
+  phase = "admission"
+  date = { ref = "admissionDateHierarchy" }
+
+  [observation.is_present]
+    field = "conjunctivitis_hx"
+    description = "Conjunctivitis"
+    ref = "Y/N"
+
+[[observation]]
+  name = "cough"
+  phase = "admission"
+  date = { ref = "admissionDateHierarchy" }
+
+  [observation.is_present]
+    field = "cough_hx"
+    description = "Cough"
+    ref = "Y/N"
+
+[[observation]]
+  name = "cough_with_sputum_production"
+  phase = "admission"
+  date = { field = "sputum_date" }
+
+  [observation.is_present]
+    field = "sputum_hx"
+    ref = "Y/N"
+
+[[observation]]
+  name = "diarrhoea"
+  phase = "admission"
+  date = { ref = "admissionDateHierarchy" }
+
+  [observation.is_present]
+    field = "diarrhoea_hx"
+    ref = "Y/N"
+
+[[observation]] # check if this is correct?!
+  name = "diastolic_blood_pressure_mmHg"
+  phase = { field = "admission", if = { redcap_event_name = "on_admission_arm_1" } }
+  date = { ref = "admissionDateHierarchy" }
```

Comment:
Do you mean the phase is admission if the name matches, otherwise it's something else? Regardless, the 'if' statement needs to go outside the 'phase' variable. You'll need something like 

phase = "admission"
    if.all = [
      { redcap_event_name = "on_admission_arm_1" },
      { "admission_diabp_vsorres" = { "!=" = "" } },
    ]

Repeat for all instances! (but check to change the name of the second if statement)

Reply:
@pipliggins - Thanks! "second if statement?"